obspy/signal/tests/test_spectral_estimation.py

```diff
@@ -677,6 +677,32 @@ def test_ppsd_temporal_plot(self):
             ppsd.plot_temporal([0.1, 1, 10], filename=ic.name, show=False,
                                **restrictions)
 
+    def test_exclude_last_sample(self):
+
+        header = {
+            "starttime":  UTCDateTime("2017-01-01T00:00:00"),
+            "network": "GR",
+            "station": "FUR",
+            "channel": "BHZ"
+        }
+
+        # 49 segments of 30 minutes to allow 30 minutes overlap in next day
+        tr = Trace(data=np.arange(30 * 60 * 49, dtype=np.int32), header=header)
```

Comment:
not sure how long this test runs.. but maybe using a smaller number of slices (3? 4?) is equally fit as a regression test? the problem should show right away after a single slice, I guess?

Reply:
Good point! I will do three segments.